src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/network/v2/OpenStackNetworkPluginTest.java

```diff
@@ -328,6 +260,265 @@ public void testSetDnsList() {
         Assert.assertEquals(dnsTwo, this.openStackNetworkPlugin.getDnsList()[1]);
     }
 
+    //test case: check if the method makes the expected calls
+    @Test
+    public void testCreateNetwork() throws FogbowException, HttpResponseException{
+        //setup
+        PowerMockito.mockStatic(CreateNetworkResponse.class);
+        Mockito.doReturn(TestUtils.EMPTY_STRING).when(openStackHttpClient).doPostRequest(Mockito.any(), Mockito.any(), Mockito.any());
+        //exercise
+        openStackNetworkPlugin.createNetwork(FAKE_NAME, openStackV3User, FAKE_TENANT_ID);
+        //verify
+        PowerMockito.verifyStatic(CreateNetworkResponse.class, Mockito.times(TestUtils.RUN_ONCE));
+        CreateNetworkResponse.fromJson(TestUtils.EMPTY_STRING);
+        Mockito.verify(openStackHttpClient, Mockito.times(TestUtils.RUN_ONCE)).doPostRequest(Mockito.any(), Mockito.any(), Mockito.any());
+    }
+
+    //test case: check if the method throws an InvalidParameterException when a JsonSyntaxException occurs.
+    @Test(expected = InvalidParameterException.class)//verify
+    public void testCreateNetworkWhenJsonException() throws FogbowException, HttpResponseException{
+        //setup
+        PowerMockito.mockStatic(CreateNetworkResponse.class);
+        PowerMockito.when(CreateNetworkResponse.fromJson(Mockito.any())).thenThrow(new JsonSyntaxException(TestUtils.EMPTY_STRING));
+        Mockito.doReturn(TestUtils.EMPTY_STRING).when(openStackHttpClient).doPostRequest(Mockito.any(), Mockito.any(), Mockito.any());
+        //exercise
+        openStackNetworkPlugin.createNetwork(FAKE_NAME, openStackV3User, FAKE_TENANT_ID);
+    }
+
+    //test case: check if the method makes the expected calls when a HttpException is thrown.
+    @Test
+    public void testCreateNetworkWhenHttpException() throws FogbowException, HttpResponseException{
+        //setup
+        PowerMockito.mockStatic(CreateNetworkResponse.class);
+        PowerMockito.mockStatic(OpenStackHttpToFogbowExceptionMapper.class);
+        HttpResponseException httpResponseException = new HttpResponseException(500, TestUtils.EMPTY_STRING);
+        Mockito.doThrow(httpResponseException).when(openStackHttpClient).doPostRequest(Mockito.any(), Mockito.any(), Mockito.any());
+
+        try {
+            //exercise
+            openStackNetworkPlugin.createNetwork(FAKE_NAME, openStackV3User, FAKE_TENANT_ID);
```

Comment:
Shouldn't the _Assert.fail_ verification method exist in this test?

Reply:
tests testCreateSubnetWhenHttpException(), testCreateSecurityGroupWhenHttpException() are looking the same as this one.